src/Test/L0/Worker/VariablesL0.cs

```diff
@@ -755,6 +755,96 @@ public void Set_StoresValue()
             }
         }
 
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public void IsReadOnly_RespectsSystemVariables()
+        {
+            using (TestHostContext hc = new TestHostContext(this))
+            {
+                // Arrange.
+                List<string> warnings;
+                var variables = new Variables(hc, new Dictionary<string, VariableValue>(), out warnings);
+                variables.Set(Constants.Variables.Agent.ReadOnlyVariables, "true");
+                variables.Set(Constants.Variables.System.AccessToken, "abc");
+                variables.Set(Constants.Variables.Agent.BuildDirectory, "abc");
+                variables.Set(Constants.Variables.Build.RepoClean, "abc");
+                variables.Set(Constants.Variables.Common.TestResultsDirectory, "abc");
+
+                // Assert.
+                Assert.True(variables.IsReadOnly(Constants.Variables.System.AccessToken));
+                Assert.True(variables.IsReadOnly(Constants.Variables.Agent.BuildDirectory));
+                Assert.True(variables.IsReadOnly(Constants.Variables.Build.RepoClean));
+                // Should be false since its not prefixed by system/agent/build
+                Assert.False(variables.IsReadOnly(Constants.Variables.Common.TestResultsDirectory));
+            }
+        }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Worker")]
+        public void IsReadOnly_RespectsUserReadOnlyVariables()
+        {
+            using (TestHostContext hc = new TestHostContext(this))
+            {
+                // Arrange.
+                List<string> warnings;
+                var variables = new Variables(hc, new Dictionary<string, VariableValue>(), out warnings);
+                variables.Set(Constants.Variables.Agent.ReadOnlyVariables, "true");
+                variables.Set("var1", "abc", false, true);
```

Comment:
nit: named parameter would be helpful to see what the true/false are specifying

Reply:
Good call